partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rex/ExprStructPermissive.kt

```diff
@@ -0,0 +1,37 @@
+package org.partiql.eval.internal.operator.rex
+
+import org.partiql.eval.internal.Environment
+import org.partiql.eval.internal.operator.Operator
+import org.partiql.eval.value.Datum
+import org.partiql.types.PType
+
+internal class ExprStructPermissive(private val fields: List<ExprStructField>) : Operator.Expr {
+    override fun eval(env: Environment): Datum {
+        val fields = fields.mapNotNull {
+            val key = it.key.eval(env)
+            if (key.isNull) {
+                return Datum.nullValue()
+            }
+            val keyString = key.getTextOrNull() ?: return Datum.struct(emptyList())
```

Comment:
Not sure this logic is correct. Consider the struct constructor expression call in permissive mode: 
```
{100: 1, 'b': 2, 'c': 3}
```

As it's currently written, this would result in an empty struct rather than omitting that single invalid struct field.